bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/categorylist/FeaturedCategoryListImpl.java

```diff
@@ -164,10 +172,15 @@ private void initModel() {
 
         List<CategoryTree> categories = categoriesRetriever.fetchCategories();
         for (CategoryTree category : categories) {
-            Map<String, String> params = new ParamsBuilder()
+            ParamsBuilder paramsBuilder = new ParamsBuilder()
                 .id(category.getId().toString())
-                .urlPath(category.getUrlPath())
-                .map();
+                .urlPath(category.getUrlPath());
+
+            if (enableUIDSupport || categoryIdentifierType == CategoryIdentifierType.UID) {
```

Comment:
Same as above, if `uid` is used we should not add `id` to the params.

Reply:
At this point we don't know. It depends on the configuration of the URLProvider.

Reply:
Arg, yes you are right we must query all options used by the URLProvider.

Reply:
Maybe the URLProvider should provide a method to return how the configuration is and what attribute is needed, this can then be used by the query.